crates/win-api-wrappers/src/service.rs

```diff
@@ -0,0 +1,142 @@
+use windows::core::Owned;
+use windows::Win32::Foundation::{ERROR_INSUFFICIENT_BUFFER, GENERIC_READ};
+use windows::Win32::System::Services::{
+    OpenSCManagerW, OpenServiceW, QueryServiceConfigW, QueryServiceStatus, StartServiceW, QUERY_SERVICE_CONFIGW,
+    SC_HANDLE, SC_MANAGER_ALL_ACCESS, SERVICE_ALL_ACCESS, SERVICE_AUTO_START, SERVICE_BOOT_START, SERVICE_DEMAND_START,
+    SERVICE_DISABLED, SERVICE_QUERY_CONFIG, SERVICE_QUERY_STATUS, SERVICE_RUNNING, SERVICE_STATUS,
+    SERVICE_SYSTEM_START,
+};
+
+use crate::utils::WideString;
+use crate::Error;
+
+pub struct ServiceManager {
+    handle: Owned<SC_HANDLE>,
+}
+
+impl ServiceManager {
+    pub fn open_read() -> Result<Self, Error> {
+        Self::open_with_access(GENERIC_READ.0)
+    }
+
+    pub fn open_all_access() -> Result<Self, Error> {
+        Self::open_with_access(SC_MANAGER_ALL_ACCESS)
+    }
+
+    fn open_with_access(access: u32) -> Result<Self, Error> {
+        // SAFETY: No preconditions.
+        let raw_sc_handle = unsafe { OpenSCManagerW(None, None, access)? };
```

Comment:
nitpick: Most of the time "No preconditions" is not true. The following may be used to declare that this is just a FFI call with the usual FFI business preconditions (valid pointers, etc).

```rust
// SAFETY: FFI call with no outstanding preconditions.
```

**If the function really have no precondition at all, it should be possible to wrap it as-is as a safe function.**

It’s not possible for OpenSCManagerW where the two first parameters are pointers. As per the documentation, it’s safe to call the function with NULL (and that’s why the windows crate is using `Option`) for that. It’s obvious enough for using the above safety comment.